calico/felix/fetcd.py

```diff
@@ -334,6 +335,56 @@ def __init__(self, config, etcd_api, status_reporter, hosts_ipset):
         self.last_rate_log_time = monotonic_time()
         # Register for events when values change.
         self._register_paths()
+        self._usage_report_greenlet = gevent.Greenlet(
+            self._periodically_usage_report
+        )
+
+    def estimated_host_count(self):
+        """
+        Single operations to python's built-in datastructures are thread-safe.
```

Comment:
This should be a comment, not in the docstring.  It's also a little misleading: this file runs under gevent, so the thing that matters is whether gevent monkey patches len() or not (I'm pretty sure it doesn't).